modules/api/pkg/resources/machine/common.go

```diff
@@ -619,6 +618,16 @@ func GetKubevirtProviderConfig(cluster *kubermaticv1.Cluster, nodeSpec apiv1.Nod
 		},
 	}
 
+	if dc.Spec.Kubevirt.EnableDedicatedCPUs {
```

Comment:
Based on https://github.com/kubermatic/dashboard/pull/7252/files#diff-1c943cc76ece28ac24a9cbdb428ae07c785ad35b82d654e0febb0b93db03f2e3R364-R368, this can be highly problematic if I enable `EnableDedicatedCPUs` for a seed which already has clusters/machinedeployments configured with "CPU". For any update(post/put) request for MD, API would start converting the CPU fields to VCPUs and thus configure both CPU and VCPUs fields which should return a validation failure from MC webhooks. 

We'll have the same case if `EnableDedicatedCPUs` is disabled with MDs present.

Reply:
I don't believe this problematic especially that we will have a documentation about this field stating that, if this field is enabled MachineDeployemtns must be rotated so it's fine at this point

Reply:
These sorts of changes should be covered by code, not some special documentation. We have not mentioned this in the [description of the field](https://github.com/kubermatic/kubermatic/blob/main/sdk/apis/kubermatic/v1/datacenter.go#L826-L828), that after enabling this, all your machine deployments need to be rotated.

Adding special cases for one customer that wouldn't really work for anyone else doesn't seem like the right way to go.

Reply:
Again, same explanation here! NamespacedMode is event further more dangerous than this and yet there are no warnings in the UI

Reply:
CPU is a field in the UI that is being influenced by this flag. What does NamespacedMode has anything to do with the UI? Not sure what your argument here is, and I can't spend more time on this. PR has been approved in whatever state it was.